packages/story-editor/src/app/media/utils/useUploadVideoFrame.ts

```diff
@@ -54,20 +69,15 @@ function useUploadVideoFrame({ updateMediaElement }) {
   const uploadVideoPoster = useCallback(
     /**
      *
-     * @param {number} id Video ID.
-     * @param {string} fileName File name.
-     * @param {File} posterFile File object.
-     * @return {Promise<{posterHeight: *, posterWidth: *, poster: *, posterId: *}>} Poster information.
+     * @param id Video ID.
+     * @param posterFile File object.
+     * @return Poster information.
      */
-    async (id, fileName, posterFile) => {
+    async (id: ResourceId, posterFile: File | null) => {
       // TODO: Make param mandatory; don't allow calling without.
       if (!posterFile) {
         return {};
       }
```

Comment:
Might be a good opportunity to make `posterFile` mandatory.

Otherwise at least we should make the return type conditional in the meantime, e.g.


```suggestion
    async (id: ResourceId, posterFile: File | null): posterFile extends File ? VideoPosterType : Record<string, never> => {
      // TODO: Make param mandatory; don't allow calling without.
      if (!posterFile) {
        return {};
      }
```